### PR TITLE
Initial port to python

### PR DESCRIPTION
**Pull request by scm-aiml**
First stage of porting to python. This focused on maintaining near mirrored implementation as typescript with some updates for more pythonic syntax when needed. Project structure also setup with poetry. 

Updated / ported files:
- browser.ts/py
- utils.ts/py

Simple unit testing also added. 

Note: Currently did not implement any ansynchorouns functionality, as we don't have a full implementation to test with a browser. It is intended that this will be implemented in future PRs.

### CODE REVIEW

File: python/src/cerebellum/browser.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+class BrowserGoalState(Enum):
+    """Enumeration of browser automation states."""
+
+    INITIAL = auto()
+    RUNNING = auto()
+    SUCCESS = auto()
+    FAILED = auto()
+
+
+@dataclass
```
Comment (theredsix) [inline python/src/cerebellum/browser.py:30]: Love the use of dataclasses for Typescript interfaces.

File: python/src/cerebellum/browser.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+from typing import Any, Literal
+
+from selenium.webdriver import ActionChains
+from selenium.webdriver.common.by import By
+from selenium.webdriver.remote.webdriver import WebDriver
+
+from .utils import parse_xdotool, pause_for_input
+
+
+class BrowserGoalState(Enum):
```
Comment (theredsix) [inline python/src/cerebellum/browser.py:21]: I'd recommend making each enum a string

**Review by theredsix — COMMENTED** [review]
Definitely a clean start!

File: python/src/cerebellum/browser.py (hunk 2 above)
Comment (scm-aiml) [reply python/src/cerebellum/browser.py]: Updated in latest commit to StrEnum

**Comment by scm-aiml** [comment]
@theredsix all of the base code is up and amazon example is running, but I just need to take care of some of the JSON serialization issues im running into on the python side with BrowserAction. Right now I converted the @dataclass objects to immutable to take care of them. But may need a custom serializer for these more complex functions.

In a future PR we may want to consider pydantic models for some of the more data class style objects.

**Comment by scm-aiml** [comment]
https://github.com/user-attachments/assets/f8b55647-9f06-4cef-81a2-ea72f4cc6e6b

**Pull request merged**